src/orion/benchmark/assessment/base.py

```diff
@@ -25,24 +25,38 @@ class BenchmarkAssessment(ABC):
     def __init__(self, task_num, **kwargs):
         self.task_number = task_num
         self._param_names = kwargs
-        self._param_names["task_num"] = task_num
+
+        self._param_names["task_num"] = (
+            int(task_num / len(kwargs.get("n_workers")))
+            if kwargs.get("n_workers")
+            else task_num
+        )
 
     @property
     def task_num(self):
         """Return the task number to run for this assessment"""
         return self.task_number
 
+    def executor(self, task_index):
```

Comment:
```suggestion
    def get_executor(self, task_index):
```
I think that would be less confusing. I was first thinking that study.executor() was executing some work rather than returning the executor.